installer/win/WixSharpInstaller/Symphony.cs

```diff
@@ -258,11 +273,68 @@ public static ActionResult UpdateConfig(Session session)
     // the requested replacement performed.
     static string ReplaceProperty( string data, string name, string value )
     {
-        // Using regular expressions to replace the existing value in the config file with the 
+        // Using regular expressions to replace the existing value in the config file with the
         // one from the property. This is the same as the regex we used to have in the old
         // Advanced Installer, which looked like this: "url"\s*:\s*".*" => "url": "[POD_URL]"
         return System.Text.RegularExpressions.Regex.Replace(data, @"""" + name + @"""\s*:\s*"".*""",
             @"""" + name + @""":""" + value.Trim() + @"""");
     }
 
+    // CleanRegistry custom action
+    [CustomAction]
+    public static ActionResult CleanRegistry(Session session)
+    {
+        try
+        {
+            // Remove registry keys added for auto-launch
+
+            using( var key = Registry.CurrentUser.OpenSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteValue("Symphony", false);
+                    key.DeleteValue("com.symphony.electron-desktop", false);
+                    key.DeleteValue("electron.app.Symphony", false);
+                }
+            }
+            using( var key = Registry.Users.OpenSubKey(@"\.DEFAULT\Software\Microsoft\Windows\CurrentVersion\Run", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteValue("Symphony", false);
+                }
+            }
+
+
+            // Remove registry keys added by protocol handlers
+
+            using( var key = Registry.CurrentUser.OpenSubKey(@"Software\Classes", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteSubKeyTree("symphony", false);
+                }
+            }
+            using( var key = Registry.LocalMachine.OpenSubKey(@"Software\Classes", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteSubKeyTree("symphony", false);
+                }
+            }
+            using( var key = Registry.ClassesRoot.OpenSubKey(@"\", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteSubKeyTree("symphony", false);
+                }
+            }
+        }
+        catch (System.Exception e)
+        {
+            session.Log("Error executing CleanRegistry: " + e.ToString() );
+            return ActionResult.Success;
```

Comment:
can we use finally and return the result in one step?

Reply:
yeah, but I think it is cleaner to do it like this, and more in line with the other custom actions which follows this pattern (as they can return ActionResult.Error)

Reply:
ok. It's a different way of coding anyway.